addons/event_crm_sale/models/event_crm.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+class EventCrm(models.Model):
+    _inherit = "event.crm"
+
+    def _get_additional_values(self, values, registration):
+        values = super(EventCrm, self)._get_additional_values(values, registration)
+        values.update({
+            'campaign_id': registration.sudo().campaign_id.id,
+            'source_id': registration.sudo().source_id.id,
+            'medium_id': registration.sudo().medium_id.id,
```

Comment:
I think the sudo thing should not be done here but should be done by caller instead. It's the responsibility of the caller to know if the user should have access or not to the models in certains conditions

Reply:
done